subprojects/docs/src/samples/workerApi/noIsolation/build.gradle

```diff
@@ -0,0 +1,50 @@
+// START SNIPPET unit-of-work
+import javax.inject.Inject
+
+// The implementation of a single unit of work
+class UnitOfWork implements Runnable {
+    File fileToReverse
+    File destinationFile
+
+    @Inject
+    public UnitOfWork(File fileToReverse, File destinationFile) {
+        this.fileToReverse = fileToReverse
+        this.destinationFile = destinationFile
+    }
+
+    @Override
+    public void run() {
+        destinationFile.text = fileToReverse.text.reverse()
+    }
+}
+// END SNIPPET unit-of-work
+
+// START SNIPPET task-implementation
+class ReverseFiles extends SourceTask {
+    @OutputDirectory
+    File outputDir
+
+    // The WorkerExecutor will be injected by Gradle at runtime
+    @Inject
+    WorkerExecutor getWorkerExecutor() {
+        throw new UnsupportedOperationException()
+    }
+
+    @TaskAction
+    void reverseFiles() {
+        // Create and submit a unit of work for each file
+        source.files.each { file ->
+            workerExecutor.submit(UnitOfWork.class) { WorkerConfiguration config ->
+                config.isolationMode = IsolationMode.NONE
+                // Constructor parameters for the unit of work implementation
+                config.params = [ file, project.file("${outputDir}/${file.name}") ]
+            }
+        }
```

Comment:
I've added a blurb about this to the `await()` javadoc, and there is already a prominent note about this in the user guide section.  If you feel like it still needs more treatment, let me know what you suggest.